### ISSUE TITLE
OSX - MultiMiner abnormal exit code 127

### ISSUE THREAD

**Issue by Merlz** [nwoolls/MultiMiner#239]
Hi,
I'm running OSX 10.11.2 and have the latest versions of Mono and XQuartz installed. 
Mono:

<pre>
Mono JIT compiler version 4.2.1 (explicit/6dd2d0d Tue Nov 10 18:01:18 EST 2015)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
    TLS:           normal
    SIGSEGV:       altstack
    Notification:  kqueue
    Architecture:  x86
    Disabled:      none
    Misc:          softdebug 
    LLVM:          yes(3.6.0svn-mono-(detached/a173357)
    GC:            sgen
</pre>

XQuartz About:

<pre>
XQuartz 2.7.8 (xorg-server 1.16.4)
</pre>

When I open MultiMiner.app I get an error in Console of:

<pre>
12/19/15 11:27:47.995 AM com.apple.xpc.launchd[1]: (MultiMiner.Win.267232[830]) Service exited with abnormal code: 127
</pre>

I am able to open it using terminal with:
mono /Applications/MultiMiner.app/Contents/Resources/MultiMiner.Win.exe

But if I click on any of the other left side icons (dashboard, Metrics, API Console etc) then it closes the window. However, I am able to click and open the top menu without issue (Start, Settings, Pools, Strategies, Perks).

When I use the command line to launch MultiMiner, the following is displayed:

<pre>
2015-12-19 11:30:44.467 mono[855:33072] WARNING: CFSTR("\37777777742\37777777634\37777777626") has non-7 bit chars, interpreting using MacOS Roman encoding for now, but this will change. Please eliminate usages of non-7 bit chars (including escaped characters above \177 octal) in CFSTR().
</pre>

Looks like a great app, if I could get it run correctly.
Thanks,
John

**Comment by tofra** [comment]
Hi John,
I have the same issue, but the error you show isn't making the app crash. This is the problem:

```
libgluezilla not found. To have webbrowser support, you need libgluezilla installed
System.NullReferenceException: Object reference not set to an instance of an object
  at System.Windows.Forms.WebBrowser.Navigate (System.String urlString) [0x00019] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.WebBrowser:Navigate (string)
  at MultiMiner.MobileMiner.Embed.WebBrowserProvider.GetWebBrowser (System.String controller, System.String emailAddress, System.String applicationKey) [0x00099] in <b64b9dd47ac24eb8bc2a980ff9a10fd5>:0
  at MultiMiner.Win.Forms.MinerForm.ShowWebBrowser (System.String controller) [0x00021] in <74af2be61f3444caa93940f28920ed5f>:0
  at MultiMiner.Win.Forms.MinerForm.dashboardButton_Click (System.Object sender, System.EventArgs e) [0x00007] in <74af2be61f3444caa93940f28920ed5f>:0
  at System.Windows.Forms.ToolStripItem.OnClick (System.EventArgs e) [0x0001c] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.ToolStripButton.OnClick (System.EventArgs e) [0x0001a] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.ToolStripItem.HandleClick (System.Int32 mouse_clicks, System.EventArgs e) [0x00036] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.ToolStripItem.FireEvent (System.EventArgs e, System.Windows.Forms.ToolStripItemEventType met) [0x00061] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.ToolStripItem:FireEvent (System.EventArgs,System.Windows.Forms.ToolStripItemEventType)
  at System.Windows.Forms.ToolStrip.OnMouseUp (System.Windows.Forms.MouseEventArgs mea) [0x00057] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.Control.WmLButtonUp (System.Windows.Forms.Message& m) [0x0007e] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.Control.WndProc (System.Windows.Forms.Message& m) [0x0016f] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.ScrollableControl.WndProc (System.Windows.Forms.Message& m) [0x00000] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.ToolStrip.WndProc (System.Windows.Forms.Message& m) [0x00000] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.Control+ControlWindowTarget.OnMessage (System.Windows.Forms.Message& m) [0x00000] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x0000b] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
  at System.Windows.Forms.NativeWindow.WndProc (System.IntPtr hWnd, System.Windows.Forms.Msg msg, System.IntPtr wParam, System.IntPtr lParam) [0x0008e] in <d5b72d15d4f7424c8a1538e3f19ec2e3>:0
```

And looking at [http://stackoverflow.com/questions/468906/how-to-instal-libgluezilla-for-mac-os-x](url) I don't have a lot of faith it can be fixed

**Comment by nwoolls** [comment]
Try the workaround found on the Installation page:

* https://github.com/nwoolls/MultiMiner/wiki/Installation#troubleshooting

The issue you are having seems to be related to the script generated by Mono when packaging the `.app` file, see:

* https://stackoverflow.com/questions/33183502/mono-on-mac-os-x-el-capitan-lsopenurlswithrole

**Comment by nwoolls** [comment]
Resolved in v4.1.0 and v4.1.1
